src/Shared.TextureContainer/Shared.TextureContainerManager.cs

```diff
@@ -151,13 +151,18 @@ public static void Release( Token holder )
         private static Texture TextureFromBytes(byte[] texBytes, TextureFormat format, bool mipmaps)
         {
             if (texBytes == null || texBytes.Length == 0) return null;
+            var imageFormat = ImageHelper.GetContentType(texBytes);
 
             //LoadImage automatically resizes the texture so the texture size doesn't matter here
             Texture2D tex = new Texture2D(2, 2, format, mipmaps);
 
             try
             {
-                tex.LoadImage(texBytes);
+                //Only use magic numbers for custom supported image formats. Let LoadImage handle png/jpg/unknown
+                if (imageFormat == ImageHelper.ImageFormat.WebP)
+                    tex = WebP.Texture2DExt.CreateTexture2DFromWebP(texBytes, mipmaps, false, out var error);
+                else
+                    tex.LoadImage(texBytes);

```

Comment:
Maybe move all image loads to a method in `ImageHelper` so it's all self-contained?